### DISCUSSION TITLE
Subgraph Queries

### DISCUSSION THREAD

**Discussion by Sshimaninja** [yuichiroaoki/poly-flashloan-bot#44] [General]
Hello Yuichiroaoki! Great work! 

I wonder; is there any reason you chose to query protocols directly, rather than using their subgraphs on The Graph?

The reason I ask is because I have read that querying on-chain data can be spoofed and that 1inch's oneSplit can be subject to exchange  related inaccuracies.

However, I've also experienced latency with The Graph, so I don't really know which option is best, and I wonder if you've looked into this.

Thank you.

**Comment by yuichiroaoki** [comment]
Hi. Interesting suggestion!
I've thought of using the Graph because I didn't know much.
I looked it up and it seems not all dexes have their own subgraph. so I have to create one to query prcie on their protocols?

I didn't know that querying on-chain data can be spoofed.

**Reply by Sshimaninja** [reply]
I'm not sure if "spoofing" is the right term, but I was looking into creating an arbitrage bot using 1inch and ran into some suggestions that it might be prone to errors. Also, I sometimes see the route on 1inch during a swap and see absolutely insane price differentials, (like, $1000/eth on some exchange I've never heard of), so I think an arbitrage bot built on 1inch routing might be fooled and try to send off transactions using this data, but encounter failed transactions and errors. Sorry I can't provide better info. However, I do know that Uniswap and Aave both suggest queries to the subgraph as a "tamper-resistant" way to query pool/factory data.

Sorry I can't provide a link to this claim, but I thought I'd mention it in case it helps you track down issues or find a solution.

I'll update you if I can find info on it.

**Reply by yuichiroaoki** [reply]
Thank you for sharing it.
That is true. You cannot trust 1inch's price data 100%, so the bot has to check price data from other sources and the subgraph would be great.

**Comment by Sshimaninja** [comment]
I noticed you've also updated the public repo to the version without 1inch. That's great to see! But, if arbitrage is a zero-sum game, and enough people run a working, profitable bot, won't the arbitrage opportunities quickly dry up? Or, is the market big enough that this shouldn't happen? Just a thought, as I started my own bot from scratch after dissecting yours a few months ago. It's been a difficult journey, but exciting, in a way. I took a lot of inspiration from your work, so I appreciate it.

**Reply by yuichiroaoki** [reply]
Well, I don't know. If that's the case, I will look for other opportunities.
yeah no problem.
